dorpsgek_github/github.py

```diff
@@ -0,0 +1,182 @@
+)
+from gidgethub.aiohttp import GitHubAPI
+from gidgethub.sansio import accept_format
+
+from dorpsgek_github.config import (
+    GITHUB_APP_ID,
+    GITHUB_APP_PRIVATE_KEY,
+    GITHUB_APP_SECRET,
+)
+
+
+router = routing.Router()
+_github_tokens = {}
+_github_installations = {}
+
+
+# HACK - Monkeypatch a fix that is already upstream
+@classmethod
+def ratelimit_from_http(cls, headers):
+    try:
+        limit = int(headers["x-ratelimit-limit"])
+        remaining = int(headers["x-ratelimit-remaining"])
+        reset_epoch = float(headers["x-ratelimit-reset"])
+    except KeyError:
+        return None
+    else:
+        return cls(limit=limit, remaining=remaining, reset_epoch=reset_epoch)
+sansio.RateLimit.from_http = ratelimit_from_http  # noqa
+
+
+class MyGitHubAPI(GitHubAPI):
+    def __init__(self, session, *args, **kwargs):
+        self._jwt = None
+        super().__init__(session, *args, **kwargs)
+
+    async def _request(self, method, url, headers, body=b''):
+        if self._jwt:
+            headers["authorization"] = f"Bearer {self._jwt}"
+        if self._oauth_token:
+            headers["authorization"] = f"token {self._oauth_token}"
+        self._jwt = None
+        self._oauth_token = None
+
+        return await super()._request(method, url, headers, body)
+
+    async def getitem(self, url, url_vars={}, *, accept=sansio.accept_format(), jwt=None, oauth_token=None):
+        # HACK - support jwt authorization; PR#62 upstream fixes this correctly.
+        # While waiting for that to land, keep the API the same, and adopt the functionality.
+        self._jwt = jwt
+        self._oauth_token = oauth_token
+
+        return await super().getitem(url, url_vars, accept=accept)
+
+    async def post(self, url, url_vars={}, *, data, accept=sansio.accept_format(), jwt=None, oauth_token=None):
+        # HACK - support jwt authorization; PR#62 upstream fixes this correctly.
+        # While waiting for that to land, keep the API the same, and adopt the functionality.
+        self._jwt = jwt
+        self._oauth_token = oauth_token
+
+        return await super().post(url, url_vars, data=data, accept=accept)
+
+
+class GitHubAPIContext:
+    """
+    Contextmanager to make GitHub API access a bit easier.
+
+    It automatically creates a new ClientSession, and sets a generic UserAgent to all GitHub API calls.
+
+    ClientSession is released to the pool on exit.
+    """
+
+    async def __aenter__(self):
+        self._session = aiohttp.ClientSession()
+        return MyGitHubAPI(self._session, "DorpsGek/0.1")
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self._session.close()
+
+
+async def process_request(headers, data):
+    """
+    Process a single request.
+
+    This is a very minimal wrapper to the Router dispatcher.
+    """
+
+    event = sansio.Event.from_http(headers,
+                                   data,
+                                   secret=GITHUB_APP_SECRET)
```

Comment:
it's a stylistic choice, but I don't like it when parameters are unnecessarily split over several lines

Reply:
either split all parameters, or none. So if it doesn't fit on one line, give them all their own line. Something like:
`headers, data,
secret=GITHUB_APP_SECRET`
is spacial very hard to read.

Reply:
In this case it fits on a single line again, so that is an easy fix :P